src/mp_game_settings.cpp

```diff
@@ -171,23 +182,33 @@ void mp_game_settings::update_addon_requirements(const config & cfg) {
 
 	mp_game_settings::addon_version_info new_data(cfg);
 
+	std::map<std::string, addon_version_info>::iterator it = addons.find(cfg["id"].str());
 	// Check if this add-on already has an entry as a dependency for this scenario. If so, try to reconcile their version info,
 	// by taking the larger of the min versions. The version should be the same for all WML from the same add-on...
-	std::map<std::string, addon_version_info>::iterator it = addons.find(cfg["id"].str());
 	if (it != addons.end()) {
-		addon_version_info & addon = it->second;
+		addon_version_info& addon = it->second;
+
+		// an add-on can contain multiple types of content
+		// for example, an era and a scenario
+		for(const auto& item : new_data.content) {
+			addon.content.emplace_back(addon_content{ item.id, item.type });
+		}
+
+		if(new_data.required) {
+			addon.required = true;
 
-		if (new_data.version) {
-			if (!addon.version || (*addon.version != *new_data.version)) {
+			if (new_data.version) {
```

Comment:
i'm not 100% sure but i think the next few lines can be simplifies by just comparing `if(addon.version == new_data.version)`

Reply:
Would it need to be `if(addon.version && new_data.version && *addon.version == *new_data.version)`?

Reply:
That would do something else, we expect that addon.version is set if an only if new_data.version is set. And that if they set that they then match.

I think the game_config_manager actually enforces this so the error message below doesn't report a bug in the umc authors code but rater a bug in the c++ engine code.

Reply:
So basically:
```
if(!new_data.version || !addon.version)
    // log error
else if(*new_data.version != *addon.version)
    // log error
```
?

Reply:
no, `if(addon.version == new_data.version) //log error. `

addon.version is not set for mainline content.

Reply:
The two places that call `update_addon_requirements` check that `addon_id` isn't empty, so no mainline content would be considered here already.

I'm not sure how `if(addon.version == new_data.version)` is a simplified equivalent to the existing checks though.  Those are end up being:
`!new_data.version && addon.version`
`new_data.version && !addon.version`
`new_data.version && (*addon.version != *new_data.version)`

Reply:
Also, for example:
```
if(addon.version == new_data.version)
	std::cout << "hello stdout" << std::endl;
```
prints out "hello stdout" when the colosseum era and scenario are selected, which I assume wouldn't be considered an error.

Reply:
ooh yes i meant `!=` not `==`

Reply:
Updated as well.